### DISCUSSION TITLE
One class per file

### DISCUSSION THREAD

**Discussion by gordanr** [dotnet/csharplang#4141] [Language Ideas]
Looking at very good proposal #137 (File scoped namespaces), I am thinking why not to remove extra nesting in case of one class per file.

Instead of 

```
using System;

public class Person
{
   public Person()
   {
      Name = "unknown";
   }

   public Person(string name)
   {
      Name = name;
   }

   public string Name { get; }

   public override string ToString()
   {
      return Name;
   }
}
```

Maybe we can write

```
using System;

public class Person;

public Person()
{
   Name = "unknown";
}

public Person(string name)
{
   Name = name;
}

public string Name { get; }

public override string ToString()
{
   return Name;
}
```

**Comment by Unknown6656** [comment]
I do not like that syntax, as it could be confused with top-level function defintions when the line `public class Person;` is omitted.
Furthermore, I always think about partial class definition (like in C++) when using this kind of statement.

Lastly, what about:
```csharp
using System;

public record R(int a, int b)
{
    public int Sum() => a + b;
}
```
This could be transformed into:
```csharp
using System;

public record R(int a, int b);

public int Sum() => a + b;
```
Which (correct me if I'm wrong), is already a legal _syntax_ (even though `a` and `b` might not be resolved in the second example.

Or would your proposal be constraint to structs, classes, and interfaces only? (excluding records)
